src/steamship/agents/context.py

```diff
@@ -0,0 +1,39 @@
+from typing import Dict, Any, List, Optional
+
+from pydantic import BaseModel
+
+from steamship import Steamship
+from steamship.experimental.transports.chat import ChatMessage
+
+
+class ChatHistory(BaseModel):
```

Comment:
Is the assumption that this is backed by a File? Just thinking out loud:

It feels like there's value in this standing alone as a (non-abstract?) base class, but maybe having a subclass that's FileTrackedChatHistory, etc?

That might support the idea that there are different chat histories. E.g. the VectorStoreBackedChatHistory, FileChatHistory, etc.

The ChatHistory might have a set of child chatHistories it pushes any updates to